### DISCUSSION TITLE
v3.1.0

### DISCUSSION THREAD

**Discussion by marcopeix** [Nixtla/neuralforecast#1388] [Announcements]
## New features
- [FEAT] (Shapley) explanations for univariate forecast models @elephaint @marcopeix  (#1377)
- [FEAT] Any horizon prediction @elephaint @JQGoh (#1368)
- [FEAT]: Training window filtering @marcopeix (#1344)
- [FEAT] xLSTM @elephaint (#1363)

## Bug fixes
- [FIX]: torch version @marcopeix (#1380)
- [FIX] Fixes for using uv @elephaint (#1364)
- [FIX] Bugs ISQF @elephaint (#1358)
- Update BiTCN doc dropout to match code @LemuelKL (#1376)
- Adds support for static variables in predict_insample method @nasaul (#1349)
- rm unused `y_insample` arg from pytorch losses @deven367 (#1360)

## General
- Dev environment installation using uv @JQGoh (#1362)
- Enhancements @deven367 (#1371)
- Codespace setup for Neuralforecast @JQGoh (#1382)
- migrate tests from nbs to pytest @deven367 (#1353)

## Documentation
- New documentation pipeline @deven367 (#1369)

<hr /><em>This discussion was created from the release <a href='https://github.com/Nixtla/neuralforecast/releases/tag/v3.1.0'>v3.1.0</a>.</em>
